packages/stores/src/currency-registrar/unsafe-ibc.ts

```diff
@@ -0,0 +1,150 @@
+import { DenomHelper } from "@keplr-wallet/common";
+import { Hash } from "@keplr-wallet/crypto";
+import { ChainStore } from "@keplr-wallet/stores";
+import {
+  AppCurrency,
+  ChainInfo,
+  Currency,
+  IBCCurrency,
+} from "@keplr-wallet/types";
+
+type OriginChainCurrencyInfo = [
+  string, // chain ID
+  string, // coinMinimalDenom
+  { portId: string; channelId: string }[] // IBC path (configured)
+];
+
+/** Will register IBC currencies to the Osmosis chain in the chain store, without querying for IBC trace denom from IBC module.
+ *  It assumes the given IBC asset config is valid.
+ *  Use for major performance boost if working with many IBC assets (as we are on Osmosis). */
+export class UnsafeIbcCurrencyRegistrar<C extends ChainInfo = ChainInfo> {
+  /** IBC hash (ibc/XXXXX) => [chainId, coinMinimalDenom] (uatom) */
+  // include chain ID, because nothing is stopping currencies from having the same config on multiple chains
+  // example: uluna on columbus (original Terra) and uluna on phoenix (new Terra)
```

Comment:
```suggestion
/** 
    IBC hash (ibc/XXXXX) => [chainId, coinMinimalDenom, IBC path]
  
    include chain ID, because nothing is stopping currencies from having the same config on multiple chains
    example: uluna on columbus (original Terra) and uluna on phoenix (new Terra)
  */
```